plugins/Multilingual/class.multilingual.plugin.php

```diff
@@ -157,8 +157,15 @@ public function profileController_setLocale_create($Sender, $locale, $TK) {
             }
         }
 
+        // Create a whitelist of the configured locales or other acceptable variables that you could want to pass;
+        $redirectVariableWhitelist = array_merge(c('EnabledLocale'), array('changed', true, 1));
+
+        $successRedirect = $_SERVER['HTTP_REFERER'];
+        if ($appendRedirect && in_array($appendRedirect, $redirectVariableWhitelist)) {
+           $successRedirect = (stripos($_SERVER['HTTP_REFERER'], '?') === false) ? $_SERVER['HTTP_REFERER'] . "?localechange=" . $appendRedirect : $_SERVER['HTTP_REFERER'] . "&localechange=" . $appendRedirect;
+        }
         // Back from whence we came.
-        Redirect($_SERVER['HTTP_REFERER']);
+        Redirect($successRedirect);
```

Comment:
There is a convention in Vanilla of simply using a `Target` parameter to specify a redirect. Will that not work here?

That $TK in the params should get removed in the future, so adding more URL params after it is problematic.